mountpoint-s3/tests/fuse_tests/read_test.rs

```diff
@@ -115,11 +143,60 @@ where
 #[cfg(feature = "s3_tests")]
 #[test]
 fn read_flexible_retrieval_test_s3() {
-    read_flexible_retrieval_test(crate::fuse_tests::s3_session::new, "read_flexible_retrieval_test");
+    const FILES: &[&str] = &["STANDARD", "GLACIER_IR", "GLACIER", "DEEP_ARCHIVE"];
+    read_flexible_retrieval_test(
+        crate::fuse_tests::s3_session::new,
+        "read_flexible_retrieval_test",
+        FILES,
+        RestorationOptions::None,
+    );
 }
 
 #[test_case(""; "no prefix")]
 #[test_case("read_flexible_retrieval_test"; "prefix")]
 fn read_flexible_retrieval_test_mock(prefix: &str) {
-    read_flexible_retrieval_test(crate::fuse_tests::mock_session::new, prefix);
+    const FILES: &[&str] = &["STANDARD", "GLACIER_IR", "GLACIER", "DEEP_ARCHIVE"];
+    read_flexible_retrieval_test(
+        crate::fuse_tests::mock_session::new,
+        prefix,
+        FILES,
+        RestorationOptions::None,
+    );
+}
+
+#[test_case(""; "no prefix")]
+#[test_case("read_flexible_retrieval_test"; "prefix")]
+fn read_flexible_retrieval_restored_test_mock(prefix: &str) {
+    const FILES: &[&str] = &["GLACIER", "DEEP_ARCHIVE"];
+    read_flexible_retrieval_test(
+        crate::fuse_tests::mock_session::new,
+        prefix,
+        FILES,
+        RestorationOptions::RestoreAndWait,
+    );
+}
+
+// This test relies on s3's expedited object restoration, it takes 1-5 minutes to complete
```

Comment:
nit: just say explicitly that we don't test `DEEP_ARCHIVE` because it has no expedited restore available

Reply:
going to test with **DEEP_ARCHIVE** object manually, need to wait it to be restored in my bucket though :) 

also it'll be good to test with **S3 Intelligent-Tiering**, because I have a suspicion that expiry date [may not be returned](https://docs.aws.amazon.com/AmazonS3/latest/userguide/restoring-objects.html#restore-archived-objects-status) for them:
> Objects restored from S3 Intelligent-Tiering Archive Access and Deep Archive Access storage tiers do not have expiration dates and instead move back to the Frequent Access tier.

but it seems not feasible, since object is put into Archive Access [90 days](https://docs.aws.amazon.com/AmazonS3/latest/userguide/intelligent-tiering-overview.html) after last access